### PR TITLE
LookUp Event Handling: rewrite

### PR DESCRIPTION
**Pull request by pavel-2020**
[Trello](https://trello.com/c/SvRDfsGC/1671-lookup-demos-3-demos)
CP: https://github.com/DevExpress/devextreme-demos/pull/1354

### CODE REVIEW

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 1]
```diff
@@ -1 +1,9 @@
-This demo illustrates how to&nbsp;perform a&nbsp;custom action when the lookup value is&nbsp;changed. Use the **onValueChanged** event handler to&nbsp;process the new lookup value.
\ No newline at end of file
+To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) handler. This demo uses it to display the picture and text for the selected item.
+
+If you want to change event handler at runtime or attach several handlers to the 
+[valueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Events/#valueChanged) event, use the [on(eventName, eventHandler)](/Documentation/ApiReference/UI_Components/dxLookup/Methods/#oneventName_eventHandler) method to subscribe to this event. This approach is more typical of jQuery. For Angular and Vue, you can use two-way data bindings. To do this in React, create a state variable.
```
Comment (AlexJameson) [inline JSDemos/Demos/Lookup/EventHandling/description.md]: Just a reminder - you can create a separate description for each approach. Vue developers are not interested in React, Angular devs in jQuery and so on.

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 2]
```diff
@@ -1 +1,9 @@
-This demo illustrates how to&nbsp;perform a&nbsp;custom action when the lookup value is&nbsp;changed. Use the **onValueChanged** event handler to&nbsp;process the new lookup value.
\ No newline at end of file
+To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) handler. This demo uses it to display the picture and text for the selected item.
```
Comment (AlexJameson) [inline JSDemos/Demos/Lookup/EventHandling/description.md]:
```suggestion
To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display the picture and text for the selected item.
```

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 3]
```diff
@@ -1 +1,9 @@
...
+
+If you want to change event handler at runtime or attach several handlers to the 
+[valueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Events/#valueChanged) event, use the [on(eventName, eventHandler)](/Documentation/ApiReference/UI_Components/dxLookup/Methods/#oneventName_eventHandler) method to subscribe to this event. This approach is more typical of jQuery. For Angular and Vue, you can use two-way data bindings. To do this in React, create a state variable.
+
+The Lookup applies the selection instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*.
+
+To specify the DOM events after which search results should be updated, use the 
+[valueChangeEvent](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#valueChangeEvent) property. The default events are input, change, and keyup.
```
Comment (AlexJameson) [inline JSDemos/Demos/Lookup/EventHandling/description.md]:
```suggestion
To specify the <a href="https://developer.mozilla.org/en-US/docs/Web/Events" target="_blank">DOM events</a> after which search results should be updated, use the [valueChangeEvent](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#valueChangeEvent) property. The default events are *"input"*, *"change"*, and *"keyup"*.
```

File: JSDemos/Demos/Lookup/EventHandling/description.md (hunk 1 above)
Comment (RomanTsukanov) [inline JSDemos/Demos/Lookup/EventHandling/description.md]: The scenario with `on` is rare. Two-way data bindings are not about event handling. Let's remove this paragraph entirely.

```suggestion
```

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 4]
```diff
@@ -1 +1,9 @@
-This demo illustrates how to&nbsp;perform a&nbsp;custom action when the lookup value is&nbsp;changed. Use the **onValueChanged** event handler to&nbsp;process the new lookup value.
\ No newline at end of file
+To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) handler. This demo uses it to display the picture and text for the selected item.
+
+If you want to change event handler at runtime or attach several handlers to the 
+[valueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Events/#valueChanged) event, use the [on(eventName, eventHandler)](/Documentation/ApiReference/UI_Components/dxLookup/Methods/#oneventName_eventHandler) method to subscribe to this event. This approach is more typical of jQuery. For Angular and Vue, you can use two-way data bindings. To do this in React, create a state variable.
+
+The Lookup applies the selection instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Lookup/EventHandling/description.md]: We haven't mentioned selection yet, so this paragraph looks out of place. Why do we even mention this property? It's not in the demo, and it is related to the topic only barely.

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 5]
```diff
@@ -1 +1,9 @@
...
+
+To specify the DOM events after which search results should be updated, use the
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Lookup/EventHandling/description.md]: "search results"? Isn't `valueChangeEvent` specify an event on which to the change the `value` property?

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 6]
```diff
@@ -1 +1,3 @@
-This demo illustrates how to&nbsp;perform a&nbsp;custom action when the lookup value is&nbsp;changed. Use the **onValueChanged** event handler to&nbsp;process the new lookup value.
\ No newline at end of file
+To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display the picture and text for the selected item.
+
+The Lookup applies the selection instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*. In this demo, you can use a switch to toggle this property.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Lookup/EventHandling/description.md]: > In this demo, you can use a switch to toggle this property.

When will this functionality be added to the demo?

```suggestion
The Lookup applies the selected value instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*. In this demo, you can use a switch to toggle this property.
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — DISMISSED** [review]

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 7]
```diff
@@ -1 +1,3 @@
-This demo illustrates how to&nbsp;perform a&nbsp;custom action when the lookup value is&nbsp;changed. Use the **onValueChanged** event handler to&nbsp;process the new lookup value.
\ No newline at end of file
+To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display the picture and text for the selected item.
```
Comment (dmitry-eliseev-devexpress) [inline JSDemos/Demos/Lookup/EventHandling/description.md]: Where should I specify the mentioned function?

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 8]
```diff
@@ -1 +1,3 @@
...
+
+The Lookup applies the selected value instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*. This functionality is not used in this demo.
```
Comment (dmitry-eliseev-devexpress) [inline JSDemos/Demos/Lookup/EventHandling/description.md]: > If you want users to click the Apply button instead

This is not an end in itself. The thing is that the described option allows you to delay the effects of your change until a click on the "Apply" button.

> This functionality **is not used** in this demo.

This passive voice is totally avoidable from the look of it. For example:

> This demo does not show this functionality.

File: JSDemos/Demos/Lookup/EventHandling/description.md [hunk 9]
```diff
@@ -1 +1,3 @@
-This demo illustrates how to&nbsp;perform a&nbsp;custom action when the lookup value is&nbsp;changed. Use the **onValueChanged** event handler to&nbsp;process the new lookup value.
\ No newline at end of file
+To handle the Lookup component's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display the picture and text for the selected item.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/Lookup/EventHandling/description.md]:
```suggestion
To handle changes to the Lookup component's value, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display a picture and text for the selected item.
```

**Review by dirk-pieterse — DISMISSED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**